Update reconstructive to ES6 class

N0taN3rd opened the pull request:
Updated the documentation per discussion about #13

N0taN3rd committed:
update reconstructive documentation pending update to ES6 class

N0taN3rd committed:
update reconstructive documentation pending update to ES6 class

ibnesayeed commented:
Some changes in this documentation are making it inconsistent with the current code, so I would suggest you to push ES6 class conversion related changes in this PR next and then we will merge them together rather than two separate PRs. Also, edit the PR title accordingly.

N0taN3rd committed:
Rewrote reconstructive to an ES6 class
Updated service worker to account for reconstructive rewrite
Added esdoc config to generate documentation
Added local development setup
Bump version

N0taN3rd committed:
Merge branch 'reconstructiveES6Class' into updatedDocsPerReconstructiveUpdate

# Conflicts:
#	reconstructive.js

N0taN3rd committed:
re-added initialization message

N0taN3rd committed:
doc styling fix

N0taN3rd committed:
documented createBanner

N0taN3rd committed:
added contributors field to package.json

N0taN3rd committed:
Reconstructive class description use markdown syntax
removed .git for the repository link in the package.json

N0taN3rd committed:
removed @desc from class description and fixed spelling

N0taN3rd committed:
make debug false by default

ibnesayeed commented on reconstructive.js:
```diff
@@ -1,116 +1,133 @@
+     * An object of functions to check whether the request should be excluded from being rerouted.
+     * Add more members to the object to add more exclusions or modify/delete existing ones.
+     * The property name can be anything descriptive of the particular exclusion, which will be shown in debug logs.
+     * Each member function is called with the fetch event and config object as parameters.
```
The config object is not passed anymore

ibnesayeed commented on reconstructive.js:
```diff
@@ -1,116 +1,133 @@
+     */
+    this._urimRegex = new RegExp(`^${this.urimPattern.replace('<datetime>', '(\\d{14})').replace('<urir>', '(.*)')}$`);
+
+    this.debug && console.log(`${this.NAME}:${this.VERSION} initialized with supplied configs`);
```
"supplied configs" => "supplied options"

ibnesayeed commented on reconstructive.js:
```diff
@@ -119,91 +136,104 @@ var Reconstructive = (function() {
     let headers = new Headers();
-    for (let hdr of original.entries()) {
-      headers.append(hdr[0], hdr[1]);
+    for (let [k, v] of original.entries()) {
```
Should we use const here instead of let?

ibnesayeed commented on reconstructive.js:
```diff
@@ -119,91 +136,104 @@ var Reconstructive = (function() {
+   * The callback function on a successful fetch from the server.
+   * Calls the rewrite() function if the response code is 2xx.
+   * Logs the response for debugging.
+   * Returns a potentially modified response. a potentially modified response.
```
" a potentially modified response" is repeated.

ibnesayeed replied:
"Returns" should be changed to "Resolves to"

ibnesayeed commented on reconstructive.js:
```diff
@@ -214,54 +244,31 @@ var Reconstructive = (function() {
+   * Rewrites the fetched response when necessary.
+   * Potential uses are to fix certain replay issues, adding an archival banner, or modifying hyperlinks.
+   * When the showBanner config is set to true, it tries to add a banner in navigational HTML pages.
+   * Returns a potentially modified response.
```
"Returns" should be changed to "Resolves to"

ibnesayeed commented on reconstructive.js:
```diff
@@ -273,103 +280,47 @@ var Reconstructive = (function() {
-   * @return  {string}              - Potentially modified response.
+   * @param  {string} datetime - The datetime of the capture
+   * @param  {string} urir - The original urir
+   * @return {string} - The banner setup
```
"setup" => "markup"

ibnesayeed commented on reconstructive.js:
```diff
@@ -119,91 +136,104 @@ var Reconstructive = (function() {
-   * @return  {Response}       - A 503 Service Unavailable response.
+   * @param  {Response}   response - Original response object.
+   * @param  {FetchEvent} event    - The fetch event.
+   * @return {Promise<Response>}   - Potentially modified response.
```
I don't think we need a Promise here.

N0taN3rd replied:
The Promise is to avoid using a union

ibnesayeed commented on reconstructive.js:
```diff
@@ -214,54 +244,31 @@ var Reconstructive = (function() {
-   * @return  {Response}            - Potentially modified response.
+   * @param  {Response}   response - Original response object.
+   * @param  {FetchEvent} event    - The fetch event.
+   * @return {Promise<Response>}   - Potentially modified response.
```
I don't think we need a Promise here.

N0taN3rd replied:
The Promise is to avoid using a union

N0taN3rd committed:
Merge remote-tracking branch 'origin/updatedDocsPerReconstructiveUpdate' into updatedDocsPerReconstructiveUpdate

# Conflicts:
#	reconstructive.js

N0taN3rd committed:
requested changes:
- createBanner: changed setup to markup
- fetchFailure: changed returns to resolves to and removed repeated documentation
- cloneHeaders: use const in for of loop
- exclusions: removed "and config object" from documentation
- initialization message: changed configs to options
spelling fixes

N0taN3rd committed:
requested changes:
- rewrite: changed returns to resolves to in documentation

ibnesayeed commented on serviceworker.js:
```diff
@@ -11,38 +11,34 @@ importScripts('reconstructive.js');
   showBanner: true,
   bannerElementLocation: `${currentPath}/reconstructive-banner.js`,
-  urimPattern: `${currentPath}/tests/<datetime>/<urir>`
+  urimPattern: `${currentPath}/tests/<datetime>/<urir>`,
```
Remove this trailing comma to be consistent.

N0taN3rd replied:
done

ibnesayeed commented on serviceworker.js:
```diff
@@ -11,38 +11,34 @@ importScripts('reconstructive.js');
-let currentPath = self.location.href.substring(0, self.location.href.lastIndexOf('/'));
-Reconstructive.init({
+const currentPath = self.location.href.substring(0, self.location.href.lastIndexOf('/'));
+const reconstructive = new Reconstructive({
```
Please change instance name to `rc` instead of `reconstructive` to avoid any confusion with the class name. Also, it will help when changing README documentation.

N0taN3rd replied:
done

N0taN3rd committed:
requested changes:
- removed trailing commas
- renamed reconstructive to rc in serviceworker.js

ibnesayeed reviewed (approved):
There are a few small things that I will fix later.

ibnesayeed commented:
README documentation and the blog post needs to be changed accordingly.
